### ISSUE TITLE
Problem with recent update from google api translate and complex strings

### ISSUE THREAD

**Issue by gbelot2003** [Stichoza/google-translate-php#119]
Hello there. Just now I update the translate  v3.4 module according with the changes you suggested, and for simple text works fine, but now the API uses the verb GET but the main problem is that I am using a text field with html, the answer is always: resulted in a `403 Forbidden` response:
Im already using something like this to clear the code:

        $inpunt1 =  str_replace("<strong>", "",  $string);
        $inpunt2 =  str_replace("</strong>", "",  $inpunt1);
        $inpunt3 =  str_replace("<br />", "",  $inpunt2);
        $inpunt4 =  str_replace("/r", "",  $inpunt3);
        $inpunt5 =  str_replace(")", "",  $inpunt4);
        $inpunt6 =  str_replace("<p>", "",  $inpunt5);
        $inpunt7 =  str_replace("</p>", "",  $inpunt6);
        $input8 = preg_replace('/\s/', ' ', $inpunt7);
        $input9 = preg_replace('/\n/', '', $input8);

even after all this cleanning Im still getting this response from the server:

Client error: `GET https://translate.google.com/translate_a/single?client=webapp&hl=en&dt=t&sl=es&tl=en&q=DESCRIPCI%26Oacute%3BN+MACROSC%26Oacute%3BPICA%3A++%26nbsp%3B++Se+recibe+1+frasco+conteniendo+2+l%26aacute%3Bminas+con+frotis+fijados+con+alcohol+de+regi%26oacute%3Bn+sangrado+pez%26oacute%3Bn+derecho%2C+se+colorean+con+hematoxilina-eosina.++++DIAGN%26Oacute%3BSTICO%3A+PEZON+DERECHO%2C+ASPIRADO+AGUJA+FINA%2C+EXTENDIDOS+-+++%26nbsp%3B++COMENTARIO%3A&ie=UTF-8&oe=UTF-8&multires=1&otf=0&pc=1&trs=1&ssel=0&tsel=0&kc=1&tk=531074.928252` resulted in a `403 Forbidden` response:

My cuestion, is there a workaround, option or maybe another module to fix this issue, 

thanks

**Comment by gbelot2003** [comment]
Ok, if anyone comes to a problem like this, for me, html_entity_decode() solve the problem!!

**Comment by ederson13** [comment]
could you provide some more details regarding your fix ?

**Comment by gbelot2003** [comment]
Sorry for the delay, is very simple indid, in the controll, the string about to be translated, it has to be sanitize first, remember, now is a get verb. I just create a simple function before translation

    private function cleanText($string)
    {
        return trim(html_entity_decode($string, ENT_COMPAT, 'UTF-8'));
    }
then just

 $factor = $this->cleanText($muestra->body);
                $body = $translator->setSource('es')
                    ->setTarget('en')
                    ->translate($factor);

**Comment by krisnantobi** [comment]
But I still got that error. I added the _function cleanText()_ before using it in _translate()_

**Comment by frzsombor** [comment]
@gbelot2003 I've tried figuring out what was the original text you wanted to translate, and by decoding the url you pasted I found that it could have been the following:

`
DESCRIPCI&Oacute;N MACROSC&Oacute;PICA:  &nbsp;  Se recibe 1 frasco conteniendo 2 l&aacute;minas con frotis fijados con alcohol de regi&oacute;n sangrado pez&oacute;n derecho, se colorean con hematoxilina-eosina.    DIAGN&Oacute;STICO: PEZON DERECHO, ASPIRADO AGUJA FINA, EXTENDIDOS -   &nbsp;  COMENTARIO:
`

As I can see, this text contains [HTML entities](https://developer.mozilla.org/en-US/docs/Glossary/Entity) and what's really interesting, if I copy&paste the text above to the official Google Translate page, I also get a translation error on that page (which I never ever saw before). See:

![image](https://user-images.githubusercontent.com/7306657/69529474-a196f480-0f70-11ea-911a-3c7ec32c3d33.png)

It looks like this is an error with Google Translate itself, and I verified this as Google Translate fails even on a simple string if it contains any HTML entity like `Hello &nbsp; world!`.

If you expect to translate strings containing HTML entities I suggest you to decode them before passing them to the translate function by using [html_entity_decode()](https://www.php.net/manual/en/function.html-entity-decode.php) as it solves this problem.

@Stichoza I also suggest that this should be noted in the known limitations until Google fixes this issue.
